CHANGELOG.md

```diff
@@ -35,6 +35,11 @@ which use the proxied NSX-V API of advanced edge gateway for handling NAT rules
 * Added methods `Client.GetVMByHref` `Vapp.GetVAMByName` and related `GetVMById`, `GetVAMByNameOrId`
 * Deprecated methods `Client.FindVMByHREF`, `Vdc.FindVMByName`, `Vdc.FindVAppByID`, and `Vdc.FindVAppByName`
 * Added methods `Vm.GetGuestCustomizationSection` and `Vm.SetGuestCustomizationSection`  
+* Added methods `CreateNsxvFirewall()`, `UpdateNsxvFirewall()`, `GetNsxvFirewallById()`, `DeleteNsxvFirewallById()`
+which use the proxied NSX-V API of advanced edge gateway for handling firewall rules [#247](https://github.com/vmware/go-vcloud-director/pull/247)
+* Added methods `GetFwGeneralParams()`, `UpdateFwGeneralParams()` for changing global firewall settings [#247](https://github.com/vmware/go-vcloud-director/pull/247)
```

Comment:
Method names need updating after a recent commit.

BTW, is "Fw" abbreviation ours or we are mimicking vCD API? I like it more with "Firewall" as in the rules' functions, as then you can search the whole code base for "Firewall" and you'll get all occurrences (and life showed that we do that with regexps when evaluating the state of code base). To make the function names shorter we could drop the word "General" and just use, say, `UpdateFirewallConfig`. Then all firewall functions would be named consistently. Your thoughts?

CC @vbauzysvmware @dataclouder

Reply:
It is me. I was relating it to the similar functions `UpdateLb...` Can drop it easily.

Reply:
If you agree, let's use the full word "Firewall" everywhere. If not, let's use the "Fw" everywhere, but, most importantly, let's be consistent :)

Reply:
I can live with Fw as with Lb -in time you get use to it. Personally prefer longer names. `General` comes from UI and API - so woudn't so keen to Drop.

Reply:
Done.

Reply:
Thanks!

Would you also share a link to vCD API docs about the `FwGeneralParamsWithXml` type. Just want to understand where the naming comes from.

Reply:
This comes more from my thinking. I needed to separate the firewall rule configuration from the regular firewall rule. Here was the word `General` coming from. Again it all started with load balancers. The `WithXml` suffix is to flag  that it carries the raw XML part of all rules inside.

`FirewallConfigWithXml` then?

Reply:
IMHO looks cleaner and consistent, but granted you have reasons for "GeneralParams".

Reply:
`FirewallConfigWithXml` it is.